modules/core/CMakeLists.txt

```diff
@@ -145,8 +148,7 @@ ocv_create_module(${extra_libs})
 
 ocv_target_link_libraries(${the_module} PRIVATE
     "${ZLIB_LIBRARIES}" "${OPENCL_LIBRARIES}" "${VA_LIBRARIES}"
-    "${OPENGL_LIBRARIES}"
-    "${GLX_LIBRARIES}"
+    "${OPENGL_LIBRARIES}" "${OPENGL_egl_LIBRARY}"
```

Comment:
@asmorkalov Does OpenGL subsystem in OpenCV uses dynamic loading? or direct linkage?

Reply:
In a way it does both and with libGLVND [the situation](https://github.com/NVIDIA/libglvnd#architecture) is a bit more complex. But anyway, it is directly linked but the function pointers are loaded a runtime.